dask_planner/src/parser.rs

```diff
@@ -18,6 +22,189 @@ macro_rules! parser_err {
+                            unexpected
+                        )))
+                    }
+                },
+                Some(Expr::UnaryOp {
+                    op: UnaryOperator::Minus,
+                    expr,
+                }) => match &**expr {
+                    Expr::Value(Value::Number(_, false)) => SqlTypeName::BIGINT,
+                    unexpected => {
+                        return Err(py_type_err(format!(
+                            "Expected string, got {:?}",
+                            unexpected
+                        )))
+                    }
+                },
+                Some(unexpected) => {
+                    return Err(py_type_err(format!(
+                        "Expected array or scalar, got {:?}",
+                        unexpected
+                    )))
+                }
+                None => {
+                    return Err(py_type_err(
+                        "PySqlArg must contain either a standard or custom AST expression",
+                    ))
+                }
+            },
+        })
+    }
+
+    #[pyo3(name = "getSqlValue")]
+    pub fn get_sql_value(&self) -> PyResult<String> {
+        Ok(match &self.custom {
+            None => match &self.expr {
+                Some(Expr::Identifier(Ident { value, .. })) => value.to_string(),
+                Some(Expr::Value(scalar)) => match scalar {
+                    Value::Boolean(value) => {
+                        if *value {
+                            "1".to_string()
+                        } else {
+                            "".to_string()
+                        }
+                    }
+                    Value::SingleQuotedString(string) => string.to_string(),
+                    Value::Number(value, false) => value.to_string(),
+                    unexpected => {
+                        return Err(py_type_err(format!(
+                            "Expected string, got {:?}",
+                            unexpected
+                        )))
+                    }
+                },
+                Some(Expr::UnaryOp {
+                    op: UnaryOperator::Minus,
+                    expr,
+                }) => match &**expr {
+                    Expr::Value(Value::Number(value, false)) => format!("-{}", value),
+                    unexpected => {
+                        return Err(py_type_err(format!(
+                            "Expected string, got {:?}",
+                            unexpected
+                        )))
+                    }
+                },
+                unexpected => {
+                    return Err(py_type_err(format!(
+                        "Expected scalar value, got {:?}",
+                        unexpected
+                    )))
+                }
+            },
+            unexpected => {
+                return Err(py_type_err(format!(
+                    "Expected scalar value, got {:?}",
+                    unexpected
+                )))
```

Comment:
It seems like we could add a helper function for these errors to reduce duplication